portlet-parent/wicketstuff-portlet/src/main/java/org/apache/wicket/portlet/PortletSystemMapper.java

```diff
@@ -0,0 +1,94 @@
+package org.apache.wicket.portlet;
+
+import org.apache.wicket.Application;
+import org.apache.wicket.RestartResponseAtInterceptPageExceptionMapper;
+import org.apache.wicket.UrlResourceReferenceMapper;
+
+import org.apache.wicket.portlet.request.mapper.BookmarkableMapper;
+import org.apache.wicket.portlet.request.mapper.HomePageMapper;
+
+import org.apache.wicket.core.request.mapper.BufferedResponseMapper;
+import org.apache.wicket.core.request.mapper.PageInstanceMapper;
+import org.apache.wicket.core.request.mapper.ResourceReferenceMapper;
+import org.apache.wicket.request.component.IRequestablePage;
+import org.apache.wicket.request.mapper.CompoundRequestMapper;
+import org.apache.wicket.request.mapper.parameter.PageParametersEncoder;
+import org.apache.wicket.request.resource.caching.IResourceCachingStrategy;
+import org.apache.wicket.util.IProvider;
+
+/**
+ * see {@link org.apache.wicket.SystemMapper}.
+ * 
+ * PortletSystemMapper encapsulates the {@link HomePageMapper} instead of the {@link org.apache.wicket.core.request.mapper.HomePageMapper}
+ * and the {@link BookmarkableMapper} instead of the {@link org.apache.wicket.core.request.mapper.BookmarkableMapper} 
+ * 
+ * @author Konstantinos Karavitis
+ */
+public class PortletSystemMapper extends CompoundRequestMapper
+{
+	private final Application application;
+
+	/**
+	 * Constructor
+	 * 
+	 * @param application
+	 */
+	public PortletSystemMapper(final Application application)
+	{
+		this.application = application;
+		add(new PageInstanceMapper());
```

Comment:
I saw your fix on WICKET-6130 issue and I will keep the PortletSystemMapper overriding the factory methods for creating the custom mappers.